src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderPreferences/GeneralType.php

```diff
@@ -53,16 +57,35 @@ class GeneralType extends TranslatorAwareType
      */
     private $tosCmsChoices;
 
+    /**
+     * @var \PrestaShopBundle\Service\Routing\Router
+     */
+    private $router;
+
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         $defaultCurrencyIsoCode,
-        array $tosCmsChoices
+        array $tosCmsChoices,
+        Router $router
     ) {
         parent::__construct($translator, $locales);
 
         $this->defaultCurrencyIsoCode = $defaultCurrencyIsoCode;
         $this->tosCmsChoices = $tosCmsChoices;
+        $this->router = $router;
+    }
+
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        parent::buildView($view, $form, $options);
```

Comment:
This is the part I dislike the most because array is not exactly intuitive to use. Maybe have DTO "FormOptions" instead? But it could be assigned in a better way too.

Reply:
A DTO seems a good thing indeed 👍  and you could gather all the `$view->vars[...] = ...` statements inside a `transferIntoFormView()` function

```
class MyDTO
{
    public function transferToView($view)
    {
        $this->view[...] = $this->...
```